fe/src/graphql/queries/test.graphql

```diff
@@ -0,0 +1,5 @@
+query GetItem($id: String!) {
```

Comment:
I think this leaves a bad example. Creating a query for each component separately would be smarter than having a global file. 

Apollo client doc example https://www.apollographql.com/docs/react/development-testing/static-typing/

Reply:
This query will be made into a hook and then we can access it anywhere

Reply:
I guess this is fine for this query as it might be used in multiple places, but I'd still avoid making global stuff like this.

Reply:
but how will the codegen generate code?. It will be pointless then

Reply:
```tsx
const GET_ROCKET_INVENTORY = gql`
  query GetRocketInventory($year: Int!) {
    rocketInventory(year: $year) {
      id
      model
      year
      stock
    }
  }
`; // <- Define query

  const { loading, data } = useQuery<RocketInventoryData, RocketInventoryVars>(
    GET_ROCKET_INVENTORY,
    { variables: { year: 2019 } }
  ); // Use in a hook

// RocketInventoryData and RocketInventoryVars are generated types
```

I'd link to a project that uses a system like this, but all of the ones I know are from work so I can't share code from them ):

Reply:
yes this is without codegen. When we were talking about codegen I thought you and everyone else liked it, so I implemented that.
So are we using codegen or hand written stuff.
Codegen would look like this: 
![image](https://user-images.githubusercontent.com/48998818/118403935-80045e00-b679-11eb-94e9-3e03b33b0414.png)

I like codegen more because it autogenerates all the types for us and it saves a lot of time